File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int i = 0, index = 0;

        while(i < chars.length) {
            int position = i;
            while(position < chars.length && chars[position] == chars[i]){
                position++;
            }
            chars[index++] = chars[i];

            if((position - i) > 1) {
                String s = Integer.toString((position - i));
                for(char c: s.toCharArray()) {
                    chars[index++] = c;
                }
            }
            i = position;
        }

        return index;
    }
}

File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        i = 0
        startIndex = 0
        while i < len(chars):
            count = 0
            ch = chars[i]
            index = i
            while index < len(chars) and chars[index] == ch:
                count += 1
                index += 1
            chars[startIndex] = ch
            startIndex += 1
            if count > 1:
                for num in str(count):
                    chars[startIndex] = num
                    startIndex += 1
            i = index
        return startIndex

        
        return